Dockerfile

```diff
@@ -1,11 +1,20 @@
 FROM dmscid/lewis-depends:latest
 
+# Copying this separately from the rest of lewis allows the
+# pip install step to be cached until the requirements change.
+# Rebuilding is faster and doesn't require an internet connection.
+COPY requirements.txt /lewis/
+
+# Install any missing PIP requirements, clear PIP cache, and delete all compiled Python objects
+RUN pip install -r /lewis/requirements.txt && \
+    rm -rf /root/.cache/pip/* && \
+    find \( -name '*.pyc' -o -name '*.pyo' \) -exec rm -rf '{}' +
+
 # Lewis source is pulled in from current directory (note .dockerignore filters out some files)
 COPY . /lewis
 
-# Install any missing PIP requirements, clear PIP cache, and delete all compiled Python objects
+# Install lewis in-place and remove compiled Python objects
 RUN pip install -e /lewis && \
```

Comment:
I don't have strong feelings about this, but from your explanations it seems that putting the dependencies into `lewis-depends` makes things overly complicated with the constant rebuilding and all.

I'm not sure on the other hand whether involving `setup.py` makes things more complicated, but not using it would force us to bring back `lewis.py` and `lewis-control.py`.